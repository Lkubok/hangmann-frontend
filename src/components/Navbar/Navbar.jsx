import React, { Component } from "react";
import { NavLink } from "react-router-dom";
import { connect } from "react-redux";
import { setUserLogOut } from "../../actions/appActions";
import * as selectors from "../../reducers/selectors";
import PropTypes from "prop-types";

import "./Navbar.scss";

export class Navbar extends Component {
  handleClick = e => {
    if (this.props.gameId && !this.props.isFinished) e.preventDefault();
  };
  render() {
    const { gameId, isFinished } = this.props;
    return (
      <nav>
        <ul className="app-navbar">
          <li>
            <NavLink
              to="/game"
              className="nav-item"
              activeClassName="nav-item-active"
              onClick={this.handleClick}
            >
              Game
            </NavLink>
          </li>

          <li>
            <NavLink
              to="/quotes"
              className={
                gameId
                  ? isFinished
                    ? "nav-item"
                    : "nav-item nav-item-disabled"
                  : "nav-item"
              }
              activeClassName="nav-item-active"
              onClick={this.handleClick}
            >
              Quotes
            </NavLink>
          </li>

          <li>
            <NavLink
              to="/addquote"
              className={
                gameId
                  ? isFinished
                    ? "nav-item"
                    : "nav-item nav-item-disabled"
                  : "nav-item"
              }
              activeClassName="nav-item-active"
              onClick={this.handleClick}
            >
              {" "}
              AddQuote
            </NavLink>
          </li>
          <li>
            <NavLink
              to="/stats"
              className={
                gameId
                  ? isFinished
                    ? "nav-item"
                    : "nav-item nav-item-disabled"
                  : "nav-item"
              }
              activeClassName="nav-item-active"
              onClick={this.handleClick}
            >
              Stats
            </NavLink>
          </li>
          <li>
            <NavLink
              to="/about"
              className={
                gameId
                  ? isFinished
                    ? "nav-item"
                    : "nav-item nav-item-disabled"
                  : "nav-item"
              }
              activeClassName="nav-item-active"
              onClick={this.handleClick}
            >
              About
            </NavLink>
          </li>
          <li>
            {this.props.isLogged ? (
              <button
                className={
                  gameId && !isFinished
                    ? "nav-item nav-item-danger nav-item-disabled"
                    : "nav-item nav-item-danger"
                }
                onClick={() => this.props.setUserLogOut()}
                disabled={gameId && !isFinished ? true : false}
              >
                LogOut
              </button>
            ) : (
              <NavLink
                to="/user"
                className={
                  gameId
                    ? isFinished
                      ? "nav-item"
                      : "nav-item nav-item-disabled"
                    : "nav-item"
                }
                activeClassName="nav-item-active"
                onClick={this.handleClick}
              >
                Login
              </NavLink>
            )}
          </li>
        </ul>
      </nav>
    );
  }
}

const mapStateToProps = state => ({
  userName: state.appParamsReducer.userName,
  isLogged: state.appParamsReducer.isLogged,
  gameId: state.gameReducer.gameId,
  isFinished: selectors.getIsFinished(state)
});

const mapDispatchToProps = {
  setUserLogOut
};

Navbar.propTypes = {
  userName: PropTypes.string,
  isLogged: PropTypes.bool,
  gameId: PropTypes.string,
  isFinished: PropTypes.bool,
  setUserLogOut: PropTypes.func
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Navbar);
