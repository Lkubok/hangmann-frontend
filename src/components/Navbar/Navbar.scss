@import "../../scss/mixins.scss";
@import "../../scss/variables.scss";

.app-navbar {
  @include basic-app-padding;
  @include flex-cr-row;
  @include basic-app-margin;
  @include basic-nav-padding;
  background-color: $dark-color;
  color: $white-color;

  li {
    flex-basis: 10%;
    list-style: none;
    border-right: 1px solid $grey-color;
    display: flex;
    &:last-child {
      border-right: none;
    }
  }
}
.nav-item {
  color: $primary-color;
  text-decoration: none;
  flex-grow: 1;
  padding: 10px;
  transition: color 0.4s ease-in-out;
  &:hover {
    color: $secondary-color;
  }
  &-active {
    color: $white-color;
  }
}
