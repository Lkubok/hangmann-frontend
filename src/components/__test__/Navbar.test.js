import React from "react";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Navbar } from "../Navbar/Navbar";

Enzyme.configure({ adapter: new Adapter() });

describe("Navbar component", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });
});
