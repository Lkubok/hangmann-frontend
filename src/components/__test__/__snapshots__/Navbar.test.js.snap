// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Should match snapshot 1`] = `
<nav>
  <ul
    className="app-navbar"
  >
    <li>
      <NavLink
        activeClassName="nav-item-active"
        className="nav-item"
        to="/game"
      >
        Game
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="nav-item-active"
        className="nav-item"
        to="/about"
      >
        About
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="nav-item-active"
        className="nav-item"
        to="/quotes"
      >
        Quotes
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="nav-item-active"
        className="nav-item"
        to="/stats"
      >
        Stats
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="nav-item-active"
        className="nav-item"
        to="/addquote"
      >
         
        AddQuote
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="nav-item-active"
        className="nav-item"
        to="/login"
      >
        Login / Sign in
      </NavLink>
    </li>
  </ul>
</nav>
`;
